dali/operators/util/get_property.cc

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/util/get_property.h"
+
+namespace dali {
+
+DALI_SCHEMA(GetProperty)
+    .DocStr(R"code()code")
+    .NumInput(1)
+    .NumOutput(1)
+    .AddArg("key", R"code()code", DALI_STRING);
+
+
+DALI_REGISTER_OPERATOR(GetProperty, GetProperty<CPUBackend>, CPU);
+
+namespace detail {
+
+template <typename Backend, typename BatchContainer = batch_container_t<Backend>>
+struct SourceInfo : public Property<Backend> {
+  TensorListShape<> GetShape(const BatchContainer& input) override {
+    TensorListShape<> ret{input.num_samples(), 1};
+    int max_length = 0;
+    for (int i = 0; i < ret.size(); i++) {
```

Comment:
Do it need to be uniform? Maybe it can be optional?

Reply:
Done, it's not uniform anymore